### ISSUE TITLE
Error: Maximum update depth exceeded in v2.6.6

### ISSUE THREAD

**Issue by ivan-aksamentov** [negomi/react-burger-menu#316]
### Description

After upgrade to react-burger-menu v2.6.6 React emits the flowing error every time I click-away from menu:

```
react-dom.development.js:56 Uncaught Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

<truncated>

menuFactory.js:142 Element with ID 'page-wrap' not found
menuFactory.js:142 Element with ID 'outer-container' not found
```
Full log is attached in the end of this message.

I think it may be related to recent changes regarding lifecycle methods or my incorrect use of hooks or react-burger-menu.

I use only [useState()](https://github.com/ivan-aksamentov/repro-react-burger-menu-2.6.6/blob/570ef2a014f932381f633258807716117453a5e6/src/components/App.js#L9) hook and not `useContext()` as described in the example in FAQ. My understanding is that `useState()` hook alone is equivalent to the older way of maintaining state in the class-based component and the use of context is optional (is this correct?).

### Repro app
An example app reproducing the issue can be found here:
https://github.com/ivan-aksamentov/repro-react-burger-menu-2.6.6

### Steps to reproduce the issue

1. Clone and run the repro:
```bash
git clone https://github.com/ivan-aksamentov/repro-react-burger-menu-2.6.6
cd repro-react-burger-menu-2.6.6
yarn install
yarn dev

```
2. Wait `webpack-dev-server` to start and navigate to `http://localhost:3000`
3. Click "Toggle menu" button. Sidebar ("push" style) will appear.
4. Click anywhere on the page to close the sidebar. The app will crash and the error messages will appear in the dev tools console.

### Workaround

Error does not occur when downgrading to react-burger-menu v2.6.5.

### Versions

| package | version |
|-------------|-----------|
| react-burger-menu | 2.6.6 |
| react | 16.8.6 |
| babel | 7.4.0 |
| webpack | 4.29.6 |
| node | 11.13.0 |
| yarn | 1.15.2 |

### Full console log

<details>
<summary>See full console log</summary>
<p>

```
log.js:24 [HMR] Waiting for update signal from WDS...
client:79 [WDS] Hot Module Replacement enabled.
react-dom.development.js:56 Uncaught Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at invariant (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:56:15)
    at scheduleWork (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:19917:5)
    at Object.enqueueSetState (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:11170:5)
    at Menu.Component.setState (webpack:///./node_modules/react/cjs/react.development.js?:335:16)
    at Menu.toggleMenu (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:75:14)
    at Menu.toggleMenu (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:652:25)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:251:16)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:642:123)
    at commitLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:17350:22)
    at commitAllLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:18737:7)
invariant @ react-dom.development.js:56
scheduleWork @ react-dom.development.js:19917
enqueueSetState @ react-dom.development.js:11170
Component.setState @ react.development.js:335
toggleMenu @ menuFactory.js:75
toggleMenu @ react-hot-loader.development.js:652
componentDidUpdate @ menuFactory.js:251
componentDidUpdate @ react-hot-loader.development.js:642
commitLifeCycles @ react-dom.development.js:17350
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
react-dom.development.js:17118 The above error occurred in the <Menu> component:
    in Menu (created by App)
    in div (created by App)
    in div (created by App)
    in App (created by HotExportedApp)
    in AppContainer (created by HotExportedApp)
    in HotExportedApp

React will try to recreate this component tree from scratch using the error boundary you provided, AppContainer.
logCapturedError @ react-dom.development.js:17118
logError @ react-dom.development.js:17154
callback @ react-dom.development.js:18095
callCallback @ react-dom.development.js:16434
commitUpdateEffects @ react-dom.development.js:16473
commitUpdateQueue @ react-dom.development.js:16461
commitLifeCycles @ react-dom.development.js:17365
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
react-hot-loader.development.js:232 Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at invariant (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:56:15)
    at scheduleWork (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:19917:5)
    at Object.enqueueSetState (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:11170:5)
    at Menu.Component.setState (webpack:///./node_modules/react/cjs/react.development.js?:335:16)
    at Menu.toggleMenu (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:75:14)
    at Menu.toggleMenu (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:652:25)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:251:16)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:642:123)
    at commitLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:17350:22)
    at commitAllLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:18737:7)
error @ react-hot-loader.development.js:232
componentDidCatch @ react-hot-loader.development.js:2139
callback @ react-dom.development.js:18096
callCallback @ react-dom.development.js:16434
commitUpdateEffects @ react-dom.development.js:16473
commitUpdateQueue @ react-dom.development.js:16461
commitLifeCycles @ react-dom.development.js:17365
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
react-hot-loader.development.js:2146 Uncaught Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at invariant (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:56:15)
    at scheduleWork (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:19917:5)
    at Object.enqueueSetState (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:11170:5)
    at Menu.Component.setState (webpack:///./node_modules/react/cjs/react.development.js?:335:16)
    at Menu.toggleMenu (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:75:14)
    at Menu.toggleMenu (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:652:25)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:251:16)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:642:123)
    at commitLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:17350:22)
    at commitAllLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:18737:7)
invariant @ react-dom.development.js:56
scheduleWork @ react-dom.development.js:19917
enqueueSetState @ react-dom.development.js:11170
Component.setState @ react.development.js:335
toggleMenu @ menuFactory.js:75
toggleMenu @ react-hot-loader.development.js:652
componentDidUpdate @ menuFactory.js:251
componentDidUpdate @ react-hot-loader.development.js:642
commitLifeCycles @ react-dom.development.js:17350
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
react-dom.development.js:17118 The above error occurred in the <AppContainer> component:
    in AppContainer (created by HotExportedApp)
    in HotExportedApp

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
logCapturedError @ react-dom.development.js:17118
logError @ react-dom.development.js:17154
update.callback @ react-dom.development.js:18066
callCallback @ react-dom.development.js:16434
commitUpdateEffects @ react-dom.development.js:16473
commitUpdateQueue @ react-dom.development.js:16461
commitLifeCycles @ react-dom.development.js:17384
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
react-dom.development.js:56 Uncaught Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at invariant (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:56:15)
    at scheduleWork (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:19917:5)
    at Object.enqueueSetState (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:11170:5)
    at Menu.Component.setState (webpack:///./node_modules/react/cjs/react.development.js?:335:16)
    at Menu.toggleMenu (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:75:14)
    at Menu.toggleMenu (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:652:25)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-burger-menu/lib/menuFactory.js?:251:16)
    at Menu.componentDidUpdate (webpack:///./node_modules/react-hot-loader/dist/react-hot-loader.development.js?:642:123)
    at commitLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:17350:22)
    at commitAllLifeCycles (webpack:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js?:18737:7)
invariant @ react-dom.development.js:56
scheduleWork @ react-dom.development.js:19917
enqueueSetState @ react-dom.development.js:11170
Component.setState @ react.development.js:335
toggleMenu @ menuFactory.js:75
toggleMenu @ react-hot-loader.development.js:652
componentDidUpdate @ menuFactory.js:251
componentDidUpdate @ react-hot-loader.development.js:642
commitLifeCycles @ react-dom.development.js:17350
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
menuFactory.js:142 Element with ID 'page-wrap' not found
handleExternalWrapper @ menuFactory.js:142
toggleMenu @ react-hot-loader.development.js:652
applyWrapperStyles @ menuFactory.js:124
toggleMenu @ react-hot-loader.development.js:652
(anonymous) @ menuFactory.js:100
setTimeout (async)
(anonymous) @ menuFactory.js:97
callCallback @ react-dom.development.js:16434
commitUpdateEffects @ react-dom.development.js:16473
commitUpdateQueue @ react-dom.development.js:16461
commitLifeCycles @ react-dom.development.js:17365
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
menuFactory.js:142 Element with ID 'outer-container' not found
handleExternalWrapper @ menuFactory.js:142
toggleMenu @ react-hot-loader.development.js:652
applyWrapperStyles @ menuFactory.js:128
toggleMenu @ react-hot-loader.development.js:652
(anonymous) @ menuFactory.js:100
setTimeout (async)
(anonymous) @ menuFactory.js:97
callCallback @ react-dom.development.js:16434
commitUpdateEffects @ react-dom.development.js:16473
commitUpdateQueue @ react-dom.development.js:16461
commitLifeCycles @ react-dom.development.js:17365
commitAllLifeCycles @ react-dom.development.js:18737
callCallback @ react-dom.development.js:150
invokeGuardedCallbackDev @ react-dom.development.js:200
invokeGuardedCallback @ react-dom.development.js:257
commitRoot @ react-dom.development.js:18949
(anonymous) @ react-dom.development.js:20419
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:20418
performWorkOnRoot @ react-dom.development.js:20347
performWork @ react-dom.development.js:20255
performSyncWork @ react-dom.development.js:20229
interactiveUpdates$1 @ react-dom.development.js:20496
interactiveUpdates @ react-dom.development.js:2171
dispatchInteractiveEvent @ react-dom.development.js:4883
```
</p>
</details>

**Comment by negomi** [comment]
Hi @ivan-aksamentov, thanks for catching this. I've just fixed it and published v2.6.7. I upgraded to this version in your repro project and confirmed it is fixed there too.

**Comment by ivan-aksamentov** [comment]
I confirm, v2.6.7 now works as expected and React strict mode warnings are now gone too.
Thank you so much for the quick fix!
